### ISSUE TITLE
Add support for ecdsa keys using scripts/ckeygen

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8828]
|[<img alt="acabhishek942's avatar" src="https://avatars.githubusercontent.com/u/7778186?s=50" width="50" height="50">](https://github.com/acabhishek942)| @acabhishek942 reported|
|-|-|
|Trac ID|trac#8828|
|Type|enhancement|
|Created|2016-09-18 07:33:51Z|
|Branch|https://github.com/twisted/twisted/tree/8828-acabhishek942-ecdsa|

Add support for ecdsa keys using scripts/ckeygen

<details><summary>Searchable metadata</summary>

```
trac-id__8828 8828
type__enhancement enhancement
reporter__acabhishek942 acabhishek942
priority__normal normal
milestone__None None
branch__8828_acabhishek942_ecdsa 8828-acabhishek942-ecdsa
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__None None
time__1474184031350161 1474184031350161
changetime__1500829322701562 1500829322701562
version__None None
owner__Cory_Benfield__lukasaoz_____ Cory Benfield <lukasaoz@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="acabhishek942's avatar" src="https://avatars.githubusercontent.com/u/7778186?s=50" width="50" height="50">](https://github.com/acabhishek942)<a name="note_1"></a>|@acabhishek942 commented|
|-|-|

see https://github.com/twisted/twisted/pull/569

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_2"></a>|@Lukasa set owner to @acabhishek|
|-|-|

Some notes have been left in the GitHub PR.

**Comment by twisted-trac** [comment]
|[<img alt="acabhishek942's avatar" src="https://avatars.githubusercontent.com/u/7778186?s=50" width="50" height="50">](https://github.com/acabhishek942)<a name="note_3"></a>|@acabhishek942 removed owner|
|-|-|

Thank you for the review.

I addressed lukasa's feedback from the github PR. but decided to keep AES-256-CBC because  cryptography's serialization.BestAvailableEncryption uses AES-256-CBC right now and without that tests fails with unknown cipher. I am adding this back to the review queue.

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)<a name="note_4"></a>|@moshez commented|
|-|-|

Switched component.

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)<a name="note_5"></a>|@moshez set owner to @acabhishek942|
|-|-|

Reviewed on GH.

**Comment by twisted-trac** [comment]
|[<img alt="acabhishek942's avatar" src="https://avatars.githubusercontent.com/u/7778186?s=50" width="50" height="50">](https://github.com/acabhishek942)<a name="note_6"></a>|@acabhishek942 removed owner|
|-|-|

Thanks for the review.

I addressed moshez's feedback and also tweaked the script a little so that it now accepts keytypes like 'RSA', 'DsA', previously only lowercase strings were accepted, this method also mimics ssh-keygen.

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)|@Lukasa set owner to @acabhishek|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="acabhishek942's avatar" src="https://avatars.githubusercontent.com/u/7778186?s=50" width="50" height="50">](https://github.com/acabhishek942)|@acabhishek942 removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)|@Lukasa set owner to @acabhishek|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="acabhishek942's avatar" src="https://avatars.githubusercontent.com/u/7778186?s=50" width="50" height="50">](https://github.com/acabhishek942)|@acabhishek942 removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_12"></a>|@Lukasa set owner to @Lukasa<br>@Lukasa set status to `closed`|
|-|-|

In changeset 2b6b2d991450fb3f61b3efd916ee89dd8490bcbe
```
#!CommitTicketReference repository="" revision="2b6b2d991450fb3f61b3efd916ee89dd8490bcbe"
Merge 8828-acabhishek942-ecdsa: Add support for ecdsa keys using scripts/ckeygen

Author: acabhishek942
Reviewer: lukasa, moshez
Fixes: #8828
```

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_13"></a>|@adiroiban commented|
|-|-|

There is also [[#5350](https://github.com/twisted/twisted/issues/5350)](https://github.com/twisted/twisted/issues/5350) as the original but in which conch was crashing as it was failing to parse an ECDSA key.

with this ticket fixed, the #5350 issue should also be fixed as conch should no longer fail to read an ECDSA key

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_14"></a>|@rodrigc commented|
|-|-|

https://github.com/twisted/twisted/pull/569#pullrequestreview-7071610
